File: kanucontrolServer/src/main/java/com/kcserver/csv/CsvImportReport.java
package com.kcserver.csv;

import lombok.Getter;

import java.util.ArrayList;
import java.util.List;

@Getter
public class CsvImportReport {

    private int totalRows;
    private int simulated;
    private int created;
    private int skipped;

    private final List<CsvImportError> errorDetails = new ArrayList<>();

    public void setTotalRows(int totalRows) {
        this.totalRows = totalRows;
    }

    public void incrementCreated() {
        this.created++;
    }

    public void incrementSimulated() {
        simulated++;
    }

    public void setSkipped(int skipped) {
        this.skipped = skipped;
    }

    public int getErrors() {
        return errorDetails.size();
    }

    public void addError(int row, String message) {
        errorDetails.add(new CsvImportError(row, message));
    }
}

File: kanucontrolClient/src/api/types/CsvImportReport.ts
export interface CsvImportError {
  row: number;
  field?: string | null;
  value?: string | null;
  message: string;
}

export interface CsvImportReport {
  totalRows: number;
  created: number;
  simulated: number;
  skipped: number;
  errors: number;
  errorDetails: CsvImportError[];
}
